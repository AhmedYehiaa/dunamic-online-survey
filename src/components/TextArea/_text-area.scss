.text-area {
  width: 100%;
  height: 100px;
  padding: $s-size;
  border: 1px solid $white;
  border-radius: $xs-size;
  font-size: 16px;
  resize: none;
}