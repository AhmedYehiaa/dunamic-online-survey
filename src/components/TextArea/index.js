import React from 'react';
import PropTypes from 'prop-types';

const TextArea = ({ value, onChange }) => (
  <textarea
    className="text-area"
    value={value}
    onChange={onChange}
  />
);

TextArea.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

TextArea.defaultProps = {
  value: '',
};
export default TextArea;
