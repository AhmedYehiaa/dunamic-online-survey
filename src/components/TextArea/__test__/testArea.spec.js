import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../index';

test('it should render textArea component with the provided props', () => {
  const onChange = jest.fn();
  const component = renderer.create(
    <TextArea
      className="text-area"
      value="my answer"
      placeholder="please enter your feedback"
      onChange={e => onChange(e.target.value)}
    />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
