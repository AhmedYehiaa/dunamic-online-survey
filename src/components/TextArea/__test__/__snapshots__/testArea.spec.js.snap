// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render textArea component with the provided props 1`] = `
<textarea
  className="text-area"
  onChange={[Function]}
  placeholder="please enter your feedback"
  value="my answer"
/>
`;
